.pre-commit-config.yaml

```diff
@@ -16,4 +16,13 @@ repos:
     - --no-sort-keys
   - id: check-merge-conflict
   - id: check-yaml
-    exclude: codebuild-ci.yaml
+    exclude: |
+      (?x)^(
+      codebuild-ci.yaml|
+      .*dependencies\.yml$
+      )$
+- repo: https://github.com/aws-cloudformation/cfn-lint
+  rev: v0.83.3
+  hooks:
+    - id: cfn-lint
+      files: .*dependencies\.yml$
```

Comment:
The only notable change in this PR.

The new files introduced Contract Test v2 artifacts for both StateMachineAlias and StateMachineVersion, and we now use a CloudFormation template (yml) to describe the necessary dependencies to run the contract tests.

check-yaml fails on the CloudFormation yml templates with the following:
`could not determine a constructor for the tag '!Ref'`

This change ensures check-yaml excludes `dependencies.yml` files and replaces the check on these files with cfn-lint instead.